accounts/abi/bind/bind_extra_test.go

```diff
@@ -0,0 +1,111 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package bind_test
+
+import (
+	"bytes"
+	"math/big"
+	"testing"
+
+	"github.com/ava-labs/coreth/accounts/abi"
+	"github.com/ava-labs/coreth/accounts/abi/bind"
+	"github.com/ava-labs/coreth/accounts/abi/bind/backends"
+	"github.com/ava-labs/coreth/core/types"
+	"github.com/ava-labs/coreth/eth/ethconfig"
+	"github.com/ava-labs/coreth/ethclient/simulated"
+	"github.com/ava-labs/coreth/node"
+	"github.com/ava-labs/coreth/params"
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/crypto"
+)
+
+// TestGetSenderNativeAssetCall checks that the NativeAssetCall proxies the
+// caller address This behavior is disabled on the network and is only to test
+// previous behavior. Note the test uses ApricotPhase2Config.
+func TestGetSenderNativeAssetCall(t *testing.T) {
+	// pragma solidity >=0.8.0 <0.9.0;
```

Comment:
Remind me to show you my automated Solidity <-> Go bindings in my personal project! I'd be happy to open-source it if we feel like it's useful.